.github/workflows/ci.yml

```diff
@@ -100,7 +99,7 @@ jobs:
             apt:
               - libopenjp2-7
               - graphviz
-        - linux: py39-online
+        - linux: py312-online
```

Comment:
Double checking, everything else has been moved to 3.10, is this being 3.12 deliberate

Reply:
This was a random choice, I could pick 310,311,312. I just picked 312 as I was lazy.